### PR TITLE
Update Ruby SDK to latest spec

### PR DESCRIPTION
**Pull request by eltonkl**
## Description of the change

* Add `remote_was_deleted` field to relevant common models
* Add new APIs and new fields to common models
* Delete removed APIs

## Type of change
- [X] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: lib/merge_hris_client/api/bank_info_api.rb [hunk 1]
```diff
@@ -90,7 +90,6 @@ def bank_info_list_with_http_info(x_account_token, opts = {})
       query_params[:'created_after'] = opts[:'created_after'] if !opts[:'created_after'].nil?
       query_params[:'created_before'] = opts[:'created_before'] if !opts[:'created_before'].nil?
       query_params[:'cursor'] = opts[:'cursor'] if !opts[:'cursor'].nil?
-      query_params[:'employee'] = opts[:'employee'] if !opts[:'employee'].nil?
```
Comment (dan-merge) [inline lib/merge_hris_client/api/bank_info_api.rb:93]: should we be removing this?

File: lib/merge_hris_client/api/employees_api.rb [hunk 2]
```diff
@@ -228,6 +236,10 @@ def employees_list_with_http_info(x_account_token, opts = {})
       if @api_client.config.client_side_validation && x_account_token.nil?
         fail ArgumentError, "Missing the required parameter 'x_account_token' when calling EmployeesApi.employees_list"
       end
+      allowable_values = ["ACTIVE", "INACTIVE", "PENDING"]
+      if @api_client.config.client_side_validation && opts[:'employment_status'] && !allowable_values.include?(opts[:'employment_status'])
```
Comment (dan-merge) [inline lib/merge_hris_client/api/employees_api.rb:240]: Probably dont want this? seems like it'll break if a value doesnt map to our enum

File: lib/merge_hris_client/api/time_off_balance_api.rb [hunk 3]
```diff
@@ -1,196 +0,0 @@
...
-
-The version of the OpenAPI document: 1.0
-Contact: hello@merge.dev
-Generated by: https://openapi-generator.tech
-OpenAPI Generator version: 5.1.1
-
-=end
-
-require 'cgi'
-
```
Comment (dan-merge) [inline lib/merge_hris_client/api/time_off_balance_api.rb:14]: why is this deleted?

Comment (eltonkl) [reply lib/merge_hris_client/api/time_off_balance_api.rb:14]: time_off_balances_api now exists after the endpoint was renamed

File: lib/merge_hris_client/api/employees_api.rb (hunk 2 above)
Comment (eltonkl) [reply lib/merge_hris_client/api/employees_api.rb:240]: auto-generated, similarly exists in various other api files as well

File: lib/merge_hris_client/api/bank_info_api.rb (hunk 1 above)
Comment (eltonkl) [reply lib/merge_hris_client/api/bank_info_api.rb:93]: doesn't exist in docs rn, guess we don't have it anymore?

**Review by gilfeig — APPROVED** [review]

**Review by leewang0 — APPROVED** [review]
